numba/core/typing/arraydecl.py

```diff
@@ -474,14 +474,19 @@ def resolve_ravel(self, ary, args, kws):
         # Only support no argument version (default order='C')
         assert not kws
         assert not args
-        return signature(ary.copy(ndim=1, layout='C'))
+        copy_will_be_made = ary.layout != 'C'
+        readonly = not (copy_will_be_made or ary.mutable)
+        return signature(ary.copy(ndim=1, layout='C', readonly=readonly))
 
     @bound_function("array.flatten")
     def resolve_flatten(self, ary, args, kws):
         # Only support no argument version (default order='C')
         assert not kws
         assert not args
-        return signature(ary.copy(ndim=1, layout='C'))
+        # Just like in Numpy: "np.copy clears previously
+        # set WRITEABLE=False flag"
+        # Numba also sets readonly=False (b/c flatten returns a copy())
```

Comment:
I'm unclear about what this comment is saying. Is there any deviation from NumPy here, or do Numba and NumPy have similar behaviour now?

Reply:
Numba and NumPy will have identical behaviour. The comment explains why the next line has a hardcoded readonly=False  argument.

Reply:
I've now rephrased the comment.